src/hrflow_connectors/connectors/sapsuccessfactors/connector.py

```diff
@@ -262,6 +263,37 @@ def format_profile(profile: HrFlowProfile) -> t.Dict:
     return sap_profile
 
 
+def format_datetime(date_string):
+    # Extract the timestamp part (milliseconds since epoch) using regular expressions
+    match = re.search(r"\d+", date_string)
+    timestamp = int(match.group())
+
+    # Convert timestamp to datetime
+    dt = datetime.datetime.utcfromtimestamp(timestamp / 1000.0)
+
+    # Convert datetime to ISO 8601 format
+    iso_date = dt.isoformat()
+    return iso_date
```

Comment:
Est ce que tu es sur de toi ici ? 
Trois remarques : 
- Pourquoi une regexp ? Est ce que tu peux pas faire juste `timestamp = float(date_string)`  
- Est ce que tu es sur de la division par 1000 que tu fais ? Coté Python d'après la doc effectivement ca s'attend à avoir des seconds since epoch donc si SAP te renvoit des millis c'est bien ce que tu fais. Mais juste pour double check 
- Tu peux enlever de tous tes PR les commentaires qui ressemblent à ca 
```python
a = 10
# Adding 25 to a 
a = a + 25  
```
Quand tu fais `.isoformat()` c'est claire que tu converti vers ISO format. 

Le seul commentaire ici qui me semble pertinent c'est par exemple dire que tu divise par 1000 parce que la data est en millis since epoch et Python s'attend à seconds since epoch

Reply:
Voila une version plus courte ou tu as besoin d'aucun commentaire 
- Nom de la fonction te dis clairement que tu recoit de l'iso format 
- Nom de l'argument précise que millis since epoch 
```python
def to_isoformat(millis_timestamp: str) -> str:
    return datetime.utcfromtimestamp(float(millis_timestamp) / 1000).isoformat()
```

Reply:
- J'utilise une regexp parce que la chaîne de date qu'ils renvoient est une chaîne de type "Date(1492098664000)",  je dois d'abord extraire le nombre.
- Oui ils me renvoient des millis
- OK 👍

Reply:
Ok pour la regexp dans ce cas. Merci pour la réponse